packages/core/theme/core-theme-browser-internal/src/theme_service.ts

```diff
@@ -43,4 +59,27 @@ export class ThemeService {
   public stop() {
     this.stop$.next();
   }
+
+  private applyTheme(darkMode: boolean) {
+    this.stylesheets.forEach((stylesheet) => {
+      stylesheet.remove();
+    });
+    this.stylesheets = [];
+    const newStylesheets = darkMode
+      ? this.themeMetadata!.stylesheetPaths.dark
+      : this.themeMetadata!.stylesheetPaths.default;
+
+    newStylesheets.forEach((stylesheet) => {
+      this.stylesheets.push(createStyleSheet({ href: stylesheet }));
+    });
+
+    _setDarkMode(darkMode);
+    updateKbnThemeTag(darkMode);
+    this.theme$.next({ darkMode });
+  }
```

Comment:
The actual logic of applying the system theme at startup.

- inject the theme-dependant css files (`2. Per-theme css files`)
- call `_setDarkMode` from `@kbn/ui-theme` (`5. euiThemeVars`)
- update the `__kbnThemeTag__` (`3. Plugin styles`) 
- emit the new value to the `theme$` observable that will be consumed by the EUI theme provider (`1. the EUI/JSS Theming`)